core/src/main/java/com/linecorp/armeria/server/annotation/decorator/LoggingDecorator.java

```diff
@@ -83,6 +83,28 @@
      */
     float failureSamplingRate() default -1.0f;
 
+    /**
+     * Sample the requests if they are slower than the {@code slowRequestSamplingPercentile} percent
+     * of the requests.
+     */
+    float slowRequestSamplingPercentile() default -1.0f;
+
+    /**
+     * Slow request percentiles are calculated over the last {@code slowRequestSamplingWindowMilliseconds}.
+     */
+    long slowRequestSamplingWindowMilliseconds() default 10 * 60 * 1000;
+
+    /**
+     * Don't sample the requests if they are faster than the {@code slowRequestSamplingLowerBoundMilliseconds}.
+     * Should be used with {@link #slowRequestSamplingUpperBoundMilliseconds()}.
+     */
+    long slowRequestSamplingLowerBoundMilliseconds() default 0L;
+
+    /**
+     * Always sample the requests if they are slower than the {@code slowRequestSamplingUpperBoundMilliseconds}.
+     */
+    long slowRequestSamplingUpperBoundMilliseconds() default Long.MAX_VALUE;
+
```

Comment:
@trustin @ikhoon @minwoox

I would like to continue working on this if you find any value on this. I find value on adding this so there is a good observability tool coming out-of-the box with Armeria, similar to printing failures with this decorator.

For example in my company, we implemented a method which has an hardcoded upper-bound only to capture slow requests right now to work-around this.

If we would like to discuss the interface we should provide to achieve this, I am open to it. I know this might not be any priority for Armeria but I would be happy to hear from you when you are available 🙂

Reply:
I personally think the existing values look sensible which are disabled by default and set values to enable the feature.